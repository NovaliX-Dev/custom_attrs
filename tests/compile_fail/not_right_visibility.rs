mod module {
    use custom_attrs::CustomAttrs;

    #[derive(CustomAttrs)]
    #[attr(a: usize)]
    #[attr(pub b: usize)]
    pub enum Enum {
        #[attr(a = 3)]
        #[attr(b = 3)]
        Variant1,

        #[attr(a = 3)]
        #[attr(b = 5)]
        Variant2
    }
}

fn main() {
    let _a = module::Enum::Variant1.get_a();
    let _b = module::Enum::Variant1.get_b();
}