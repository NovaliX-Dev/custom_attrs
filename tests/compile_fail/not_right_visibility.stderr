error[E0624]: associated function `get_a` is private
  --> tests/compile_fail/not_right_visibility.rs:19:37
   |
4  |     #[derive(CustomAttrs)]
   |              ----------- private associated function defined here
...
19 |     let _a = module::Enum::Variant1.get_a();
   |                                     ^^^^^ private associated function
